olp-cpp-sdk-core/include/olp/core/network2/NetworkProxySettings.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <olp/core/CoreApi.h>
+
+namespace olp {
+namespace network2 {
+/**
+ * @brief The NetworkProxySettings class contains proxy configuration for the
+ * network
+ */
+class CORE_API NetworkProxySettings final {
+ public:
+  /**
+   * @brief The proxy type
+   */
+  enum class Type { NONE, HTTP, SOCKS4, SOCKS4A, SOCKS5, SOCKS5_HOSTNAME };
+
+ public:
+  /**
+   * @brief Get proxy type
+   * @return proxy type
+   */
+  Type GetType() const;
+  /**
+   * @brief Set proxy type
+   * @param[in] proxy type
+   * @return reference to *this
+   */
+  NetworkProxySettings& WithType(Type type);
+
+  /**
+   * @brief Get proxy hostname
+   * @return proxy hostname
+   */
+  const std::string& GetHostname() const;
+  /**
+   * Set proxy hostname
+   * @param[in] proxy hostname
+   * @return reference to *this
+   */
+  NetworkProxySettings& WithHostname(std::string hostname);
+
+  /**
+   * @brief Get proxy port
+   * @return proxy port
+   */
+  std::uint16_t GetPort() const;
+  /**
+   * @brief Set proxy port
+   * @param[in] proxy port
+   * @return reference to *this
+   */
+  NetworkProxySettings& WithPort(std::uint16_t port);
+
+  /**
+   * @brief Get username
+   * @return username
+   */
+  const std::string& GetUsername() const;
+  /**
+   * @brief Set username
+   * @param[in] username
+   * @return reference to *this
+   */
+  NetworkProxySettings& WithUsername(std::string username);
+
+  /**
+   * @brief Get password
+   * @return password
+   */
+  const std::string& GetPassword() const;
+  /**
+   * @brief Set password
+   * @param[in] password
+   * @return reference to *this
+   */
+  NetworkProxySettings& WithPassword(std::string password);
+
+ private:
+  Type type_ = Type::NONE;  /// proxy type
```

Comment:
I would like to omit to put comments for members after declaration.
```suggestion
  /// The type of the proxy.
  Type type_ = Type::NONE;
```
Also please notice that ///< is the proper doxygen command for _after_ member comments.